services/src/main/java/org/keycloak/authentication/authenticators/x509/ValidateX509CertificateUsername.java

```diff
@@ -76,8 +78,7 @@ public void authenticate(AuthenticationFlowContext context) {
                     .validateExtendedKeyUsage();
         } catch(Exception e) {
             logger.error(e.getMessage(), e);
-            // TODO use specific locale to load error messages
-            Response challengeResponse = errorResponse(Response.Status.UNAUTHORIZED.getStatusCode(), "invalid_request", e.getMessage());
+            Response challengeResponse = ErrorPage.error(context.getSession(), context.getAuthenticationSession(), Response.Status.UNAUTHORIZED, Messages.AUTHENTICATION_FAILED, e.getMessage());
```

Comment:
We should just use internalServerError, with 500 as the return status code here I think